bob-common/src/mapper.rs

```diff
@@ -134,15 +134,14 @@ impl Virtual {
         let mut support_nodes = Vec::with_capacity(count);
         for (id, node) in self.nodes.iter().skip(offset % self.nodes.len()) {
             if target_nodes.iter().all(|i| i.index() != *id) {
-                support_nodes.push(node);
-                if support_nodes.len() >= count {
-                    break;
+                if support_nodes.len() < count && node.connection_available() {
+                    support_nodes.push(node);
                 }
             }
         }
         if support_nodes.len() < count && offset % self.nodes.len() > 0 {
             for (id, node) in self.nodes.iter().take(offset % self.nodes.len()) {
```

Comment:
Probably, we need a single hand written circular for loop to properly implement skipping of disconnected nodes with offset